src/datasets/io/json.py

```diff
@@ -63,49 +64,89 @@ def __init__(
         dataset: Dataset,
         path_or_buf: Union[PathLike, BinaryIO],
         batch_size: Optional[int] = None,
+        num_proc: Optional[int] = None,
         **to_json_kwargs,
     ):
         self.dataset = dataset
         self.path_or_buf = path_or_buf
-        self.batch_size = batch_size
+        if batch_size:
+            self.batch_size = batch_size
+        elif num_proc is not None:
+            self.batch_size = 100_000
+        else:
+            self.batch_size = config.DEFAULT_MAX_BATCH_SIZE
```

Comment:
Why not use the same default value here ?

Reply:
I realised that each Process (In multiprocessing) is capable of handling a much higher batch size which makes it more faster as compared to current version. Same default value will give only 10% speed up. We can do that as well and mention in docstring to increase batch size? What do you think?

Reply:
It really depends on the data. Any batch size bigger than 100,000 can cause memory issues on small computers if the dataset has long examples.
IMO we should stick to the default `config.DEFAULT_MAX_BATCH_SIZE` even though it's not the most optimal for the text data you used. At least it won't crash for processed vision/audio datasets x)
A note in the docstring to explain this would be very welcome !

Reply:
Hi @bhavitvyamalik , have you had a chance to consider using `config.DEFAULT_MAX_BATCH_SIZE` by default ?

Reply:
Hey! Yeah, I've updated the docstring. Also, we need to have separate `test_dataset_to_json_lines` and `test_dataset_to_json_orient` for this right?